packages/flutter/lib/src/rendering/box.dart

```diff
@@ -1985,7 +1985,7 @@ abstract class RenderBox extends RenderObject {
       }
       return true;
     }());
-    return _size!;
+    return _size ?? (throw StateError('RenderBox was not laid out: $runtimeType#${shortHash(this)}'));
```

Comment:
We avoid calling `toString` on `runtimeType` (as is done here implicitly) in release mode, see [objectRuntimeType](https://master-api.flutter.dev/flutter/foundation/objectRuntimeType.html)

Reply:
I did it deliberately ;) This is because `objectRuntimeType` will yield `<optimized out>` at runtime, while the main idea of the PR is to know *which* exact class causes the bug. 

Given the godbolt decompile results above, it seems that, as long as we do not touch this branch (i.e. throw error), we have zero overhead. When we throw error, we already have a ton of overhead and trouble so this runtimeType.toString does not seem to be a big problem.